ui/pages/confirmations/components/confirm/info/shared/advanced-details/advanced-details.tsx

```diff
@@ -0,0 +1,174 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowText,
+} from '../../../../../../../components/app/confirm/info/row';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Icon,
+  IconName,
+} from '../../../../../../../components/component-library';
+import Tooltip from '../../../../../../../components/ui/tooltip';
+import {
+  BlockSize,
+  BorderColor,
+  IconColor,
+} from '../../../../../../../helpers/constants/design-system';
+import { useCopyToClipboard } from '../../../../../../../hooks/useCopyToClipboard';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import {
+  currentConfirmationSelector,
+  getCustomNonceValue,
+  getNextSuggestedNonce,
+  getUseNonceField,
+} from '../../../../../../../selectors';
+import {
+  getNextNonce,
+  showModal,
+  updateCustomNonce,
+} from '../../../../../../../store/actions';
+import { useKnownMethodDataInTransaction } from '../../hooks/known-method-data-in-transaction';
+
+const NonceDetails = () => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getNextNonce());
+  }, [dispatch]);
+
+  const useNonceField = useSelector(getUseNonceField);
+  const nextNonce = useSelector(getNextSuggestedNonce);
+  const customNonceValue = useSelector(getCustomNonceValue);
+
+  const openEditNonceModal = () =>
+    dispatch(
+      showModal({
+        name: 'CUSTOMIZE_NONCE',
+        useNonceField,
+        nextNonce,
+        customNonceValue,
+        updateCustomNonce: (value: string) => {
+          dispatch(updateCustomNonce(value));
+        },
+        getNextNonce,
+      }),
+    );
+
+  const displayedNonce = customNonceValue || nextNonce;
+
+  return (
+    <Box padding={2}>
+      <ConfirmInfoRow
+        label={t('advancedDetailsNonceDesc')}
+        tooltip={t('advancedDetailsNonceTooltip')}
+      >
+        <ConfirmInfoRowText
+          text={`${displayedNonce}`}
```

Comment:
Might be personal preference, but various sites and tools display nonces in hex, so a cool feature could be to also show the nonce in hexadecimal if the advanced toggle is on?

Reply:
I believe we've only considered decimal nonces in the designs, but I'm raising this in our slack channel as a potential future addition.